plugins/outputs/honeycomb/honeycomb.go

```diff
@@ -0,0 +1,133 @@
+package honeycomb
+
+import (
+	"errors"
+	"fmt"
+	"github.com/honeycombio/libhoney-go"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type Honeycomb struct {
+	APIKey      string   `toml:"apiKey"`
+	Dataset     string   `toml:"dataset"`
+	APIHost     string   `toml:"apiHost"`
+	SpecialTags []string `toml:"specialTags"`
+}
+
+var sampleConfig = `
+  ## Honeycomb authentication token
+  apiKey = "API_KEY"
+
+  ## Dataset name in Honeycomb to send data to
+  dataset = "my-dataset"  
+
+  ## Special tags that will not get prefixed by the measurement name
+  ## This should be set if you specified global tags, and it should include the list of all global tags + host
+  ## Default value for this list is: host
+  #specialTags = ["host"]
+  
+  ## Optional: the hostname for the Honeycomb API server
+  #apiHost = "https://api.honeycomb.io/""
+`
+
+func (h *Honeycomb) Connect() error {
+
+	if h.APIKey == "" {
+		return errors.New("Honeycomb apiKey can not be empty")
+	}
+
+	if h.Dataset == "" {
+		return errors.New("Honeycomb dataset can not be empty")
+	}
+
```

Comment:
These checks should be done in an `func (h *Honeycomb) Init() error` function that is called by Telegraf _after_ setting the user-configured parameters.